api/types.go

```diff
@@ -43,28 +43,34 @@ type GenerateRequest struct {
 	Options map[string]interface{} `json:"options"`
 }
 
+type ImageData struct {
+	Data string `json:"data"`
+	ID   int    `json:"id"`
+}
+
 // Options specfied in GenerateRequest, if you add a new option here add it to the API docs also
 type Options struct {
 	Runner
 
 	// Predict options used at runtime
-	NumKeep          int      `json:"num_keep,omitempty"`
-	Seed             int      `json:"seed,omitempty"`
-	NumPredict       int      `json:"num_predict,omitempty"`
-	TopK             int      `json:"top_k,omitempty"`
-	TopP             float32  `json:"top_p,omitempty"`
-	TFSZ             float32  `json:"tfs_z,omitempty"`
-	TypicalP         float32  `json:"typical_p,omitempty"`
-	RepeatLastN      int      `json:"repeat_last_n,omitempty"`
-	Temperature      float32  `json:"temperature,omitempty"`
-	RepeatPenalty    float32  `json:"repeat_penalty,omitempty"`
-	PresencePenalty  float32  `json:"presence_penalty,omitempty"`
-	FrequencyPenalty float32  `json:"frequency_penalty,omitempty"`
-	Mirostat         int      `json:"mirostat,omitempty"`
-	MirostatTau      float32  `json:"mirostat_tau,omitempty"`
-	MirostatEta      float32  `json:"mirostat_eta,omitempty"`
-	PenalizeNewline  bool     `json:"penalize_newline,omitempty"`
-	Stop             []string `json:"stop,omitempty"`
+	NumKeep          int         `json:"num_keep,omitempty"`
+	Seed             int         `json:"seed,omitempty"`
+	NumPredict       int         `json:"num_predict,omitempty"`
+	TopK             int         `json:"top_k,omitempty"`
+	TopP             float32     `json:"top_p,omitempty"`
+	TFSZ             float32     `json:"tfs_z,omitempty"`
+	TypicalP         float32     `json:"typical_p,omitempty"`
+	RepeatLastN      int         `json:"repeat_last_n,omitempty"`
+	Temperature      float32     `json:"temperature,omitempty"`
+	RepeatPenalty    float32     `json:"repeat_penalty,omitempty"`
+	PresencePenalty  float32     `json:"presence_penalty,omitempty"`
+	FrequencyPenalty float32     `json:"frequency_penalty,omitempty"`
+	Mirostat         int         `json:"mirostat,omitempty"`
+	MirostatTau      float32     `json:"mirostat_tau,omitempty"`
+	MirostatEta      float32     `json:"mirostat_eta,omitempty"`
+	PenalizeNewline  bool        `json:"penalize_newline,omitempty"`
+	Stop             []string    `json:"stop,omitempty"`
+	ImageData        []ImageData `json:"image_data,omitempty"`
```

Comment:
I'd prefer to see `ImageData` directly as an optional parameter in the body the `/generate` request personally.

Something like this:
```
curl -X POST http://localhost:11434/api/generate -d '{
    "model": "mistral",
    "prompt": "Please tell me what text is in this photo [img-1]",
    "images": [
        {
            "id": 1,
            "data": <b64-data>
        }
    ]
}'
```

cc: @jmorganca for his opinion

Reply:
Happy to do so, but used this as its 1:1 with llama.cpp

Reply:
It definitely makes more sense as a parameter in `/generate`. The id part feels _very_ foreign to me from a UX standpoint (I think a lot of people will trip over this).

Reply:
Any recommendations? I just exposed this from llama.cpp

Reply:
We could do it dynamically? indexes have to be numbers so... not hard to implement and maybe easier to just "get"?